### PR TITLE
Add machine exec docker infra implementation

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
Reference issues:
https://github.com/eclipse/che/issues/10817
https://github.com/eclipse/che/issues/10590

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

**Comment by skabashnyuk** [comment]
I saw in a lot of projects that they commit vendor folder in the source code. We do like that here https://github.com/eclipse/che-jwtproxy https://github.com/eclipse/che/tree/master/agents/go-agents  it will simplify quite a lot of things in your pr.

**Comment by AndrienkoAleksandr** [comment]
@skabashnyuk thanks, I applied vendor folder.

**Comment by garagatyi** [comment]
+1 to use `dep` to vendor dependencies

File: api/model/model_types.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package model
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	"github.com/eclipse/che-lib/websocket"
```
Comment (skabashnyuk) [inline api/model/model_types.go]: you can reference github.com/gorilla/websocket  directly here.
We have version 361d4c0ffd78338ebe0a9e6320cdbe115d7dc026 approved in https://dev.eclipse.org/ipzilla/show_bug.cgi?id=10452

Comment (AndrienkoAleksandr) [reply api/model/model_types.go]: Oh, yes. Thanks, it's a good idea.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+docker build -t eclipse/che-machine-exec .
+```
+
+# Run docker container
+
+Run docker container with che-machine-exec manually:
+
+```
+docker run --rm -p 4444:4444 -v /var/run/docker.sock:/var/run/docker.sock eclipse/che-machine-exec
+````
```
Comment (sunix) [inline README.md]: To me, it is missing 2 things:
- an architecture overview
- code samples and tutorial on how to have it working: workspace config, tutorial on how to implement a simple client side app that would use it.

Comment (AndrienkoAleksandr) [reply README.md]: Thanks. I will create separated issue for that.

Comment (AndrienkoAleksandr) [reply README.md]: Create issue: https://github.com/eclipse/che-machine-exec/issues/2

File: vendor/github.com/Microsoft/go-winio/backup.go [hunk 3]
```diff
@@ -230,10 +219,11 @@ type BackupFileWriter struct {
 	ctx             uintptr
 }
 
-// NewBackupFileWriter returns a new BackupFileWriter from a file handle. If includeSecurity is true,
+// NewBackupFileWrtier returns a new BackupFileWriter from a file handle. If includeSecurity is true,
```
Comment (mshaposhnik) [inline vendor/github.com/Microsoft/go-winio/backup.go]: Writer

Comment (mshaposhnik) [reply vendor/github.com/Microsoft/go-winio/backup.go]: Ah it's library code. Ok. Never mind :)

Comment (AndrienkoAleksandr) [reply vendor/github.com/Microsoft/go-winio/backup.go]: It's vendor code. We should not modify it.

**Pull request merged**